### DISCUSSION TITLE
"Recently Modified" - How does gerbera know?

### DISCUSSION THREAD

**Discussion by udippel** [gerbera/gerbera#3194] [General]
As a follow-up of #3188, I tried to repeat the exactly same state(s) after the application of #3192. I was not able to reproduce any error, so I think that #3188 is actually closed for good.

One question remains, however: after deleting the PC directory ("`test_dir`") which was set on autoscan, method `grb`, and a restart, gerbera came back with no files at all. Re-activated that directory with the eight `Jamaïque_x_8..` files.
What I don't understand, though: while the "`Recently Added`" shows these eight files, as to be expected; the "`Recently Modified`" show these **two**:
![image](https://github.com/user-attachments/assets/ff94f9d4-3f3f-442e-be48-e845014c1c58)
These are actually the 'most recently added' files to directory test-dir; however, well before the deletion of PC directory, and the ensuing new start, and newly added `test-dir` as autoscan. 
How does gerbera know? Does it - despite of the deletion of PC directory - still remember earlier modifications?

I have tried to reproduce this, and I can reproduce this. So far I don't see it as bug. More curiosity: after a fresh start, with a presumably empty database, files in a newly added directory are shown as "Recently Added". All good. Though: how can there be two "Recently Modified"? If at all, all eight files were recently 'modified' (ie added); or none? Or is there anything that I don't understand correctly?

**Comment by KarlStraussberger** [comment]
Recently modified is taken from the file date.

**Comment by udippel** [comment]
That's what I had  thought as possibility. 
How far back does it go? How many files maximum? Like this it may be useful, as long as the user knows what it actually teaches him.
It is not a priori logical that "Added ..." is taken from the database, while "Modified ..." comes from physical file date / touch. 
"Modified ..." could as well mean 'deleted', which I'd personally prefer.

In any case, thanks very much for the reply!

**Reply by KarlStraussberger** [reply]
It's in your config.xml

```
<containers enabled="yes">
      <container location="/LastAdded" title="Recently Added" sort="-last_updated">
        <filter>upnp:class derivedfrom "object.item" and last_updated &gt; "@last7"</filter>
      </container>
      <container location="/LastModified" title="Recently Modified" sort="-last_modified">
        <filter>upnp:class derivedfrom "object.item" and last_modified &gt; "@last7"</filter>
      </container>
    </containers>
```

The meaning of the filters is described in the docs.

**Comment by udippel** [comment]
Thanks, so it's the default, since I use the gerbera-generated config for this test install.
